hack/util.sh

```diff
@@ -23,6 +23,37 @@ function util::create_host_cluster() {
   local KUBECONFIG_PATH=${2}
   local MANIFEST_DIR=${3}
   local CONFIG_DIR=${4}
+  local HOST_CONTEXT
+
+  if [[ $CLUSTER_PROVIDER == "kind" ]]; then
+    util::create_host_kind_cluster "${HOST_CLUSTER_NAME}" "${KUBECONFIG_PATH}"
+    HOST_CONTEXT="kind-${HOST_CLUSTER_NAME}"
+  elif [[ $CLUSTER_PROVIDER == "kwok" ]]; then
+    util::create_host_kwok_cluster "${HOST_CLUSTER_NAME}" "${KUBECONFIG_PATH}"
+    HOST_CONTEXT="kwok-${HOST_CLUSTER_NAME}"
+  else
+    echo "Invalid provider, only kwok or kind allowed"
+    exit 1
+  fi
+
+  kubectl --kubeconfig="${KUBECONFIG_PATH}" --context="${HOST_CONTEXT}" create -f "${MANIFEST_DIR}"
+  kubectl --kubeconfig="${KUBECONFIG_PATH}" --context="${HOST_CONTEXT}" create -f "${CONFIG_DIR}"
+
+  NOFED=kubeadmiral.io/no-federated-resource=1
+  kubectl --kubeconfig="${KUBECONFIG_PATH}" --context="${HOST_CONTEXT}" annotate deploy -A --all $NOFED
+  kubectl --kubeconfig="${KUBECONFIG_PATH}" --context="${HOST_CONTEXT}" annotate daemonset -A --all $NOFED
+  kubectl --kubeconfig="${KUBECONFIG_PATH}" --context="${HOST_CONTEXT}" annotate cm -A --all $NOFED
+  kubectl --kubeconfig="${KUBECONFIG_PATH}" --context="${HOST_CONTEXT}" annotate secret -A --all $NOFED
+  kubectl --kubeconfig="${KUBECONFIG_PATH}" --context="${HOST_CONTEXT}" annotate role -A --all $NOFED
+  kubectl --kubeconfig="${KUBECONFIG_PATH}" --context="${HOST_CONTEXT}" annotate rolebinding -A --all $NOFED
+  kubectl --kubeconfig="${KUBECONFIG_PATH}" --context="${HOST_CONTEXT}" annotate clusterrole -A --all $NOFED
+  kubectl --kubeconfig="${KUBECONFIG_PATH}" --context="${HOST_CONTEXT}" annotate clusterrolebinding -A --all $NOFED
+  kubectl --kubeconfig="${KUBECONFIG_PATH}" --context="${HOST_CONTEXT}" annotate svc -A --all $NOFED
+}
+
+function util::create_host_kind_cluster() {
+  local HOST_CLUSTER_NAME=${1}
+  local KUBECONFIG_PATH=${2}
 
   local KIND_CONFIG
   KIND_CONFIG=$(mktemp)
```

Comment:
nit: creating a temp file may not be necessary here. 

Maybe one of the following might work?

```bash
KIND_CONFIG=(cat <<EOF
...
EOF)

kind create cluster --config <(echo "$KIND_CONFIG") ......
```

or

```bash
kind create cluster --config /dev/stdin ...... << EOF
...
EOF 
```

Reply:
just tested and the first approach seems to work, will update the script